flepimop/gempyor_pkg/src/gempyor/distributions.py

```diff
@@ -0,0 +1,434 @@
+        array([[0.3534603 , 2.11795541, 2.56142749, 0.14212687, 0.27193845],
+               [1.13637163, 0.72888261, 0.98333919, 0.42611589, 2.40944872],
+               [2.17666075, 1.06825951, 3.08712799, 1.59601411, 0.42346159]])
+    """
+
+    distribution: Literal["lognorm"] = "lognorm"
+    meanlog: float
+    sdlog: float = Field(..., gt=0)
+
+    def _sample_from_generator(
+        self, size: int | tuple[int, ...], rng: Generator
+    ) -> npt.NDArray[np.float64]:
+        """Sampling logic for lognormal distributions."""
+        return rng.lognormal(mean=self.meanlog, sigma=self.sdlog, size=size)
+
+
+class TruncatedNormalDistribution(DistributionABC):
+    # pylint: disable=line-too-long
+    """
+    Represents a truncated normal distribution.
+
+    Examples:
+        >>> import numpy as np
+        >>> from gempyor.distributions import TruncatedNormalDistribution
+        >>> rng = np.random.default_rng(42)
+        >>> dist = TruncatedNormalDistribution(mean=1.0, sd=1.0, a=0.0, b=10.0)
+        >>> dist
+        TruncatedNormalDistribution(distribution='truncnorm', mean=1.0, sd=1.0, a=0.0, b=10.0)
+        >>> dist.sample(rng=rng)
+        array([1.87722989])
+        >>> dist.sample(size=(3, 5), rng=rng)
+        array([[1.07000038, 2.18016199, 1.66002835, 0.28689654, 3.04332767],
+               [1.83818339, 1.9153892 , 0.37639339, 1.09435167, 0.92629918],
+               [2.54134935, 1.52545861, 2.04022114, 1.07959285, 0.6142512 ]])
+    """
+    # pylint: enable=line-too-long
+
+    distribution: Literal["truncnorm"] = "truncnorm"
+    mean: float
+    sd: float = Field(..., gt=0)
+    a: float
+    b: float
+
+    def _sample_from_generator(
+        self, size: int | tuple[int, ...], rng: Generator
+    ) -> npt.NDArray[np.float64]:
+        """Sampling logic for truncated normal distributions."""
+        if (
+            isclose(self.a, self.b) and self.allow_edge_cases
+        ):  # use this logic b/c scipy.truncnorm doesn't support equal bounds
+            return np.full(size, self.a)
+
+        lower = (self.a - self.mean) / self.sd
+        upper = (self.b - self.mean) / self.sd
+        return truncnorm.rvs(
+            a=lower,
+            b=upper,
+            loc=self.mean,
+            scale=self.sd,
+            size=size,
+            random_state=rng,
+        )
+
+    @model_validator(mode="after")
+    def _validate_bounds(self) -> "TruncatedNormalDistribution":
+        """Validate bounds based on whether or not edge cases are allowed.."""
+        if self.allow_edge_cases:
+            if self.b < self.a:
+                raise ValueError(
+                    f"Upper bound 'b' ({self.b}) must be ≥ to lower bound 'a' ({self.a})."
+                )
+        else:
+            if self.b < self.a or isclose(self.b, self.a):
+                raise ValueError(
+                    f"Upper bound 'b' ({self.b}) must be > lower bound 'a' ({self.a})."
+                )
+        return self
```

Comment:
```suggestion
    @model_validator(mode="after")
    def _validate_bounds(self) -> "TruncatedNormalDistribution":
        """Validate bounds based on whether or not edge cases are allowed.."""
        if self.b < self.a or (not self.allow_edge_cases and isclose(self.a, self.b)):
          op = ">=" if self.allow_edge_cases else ">"
          raise ValueError(f"Upper bound `b`, {self.b}, must be {op} to lower bound `a`, {self.a}.")
        return self
```
And similar for `UniformDistribution`.